src/components/MultiHopTrade/hooks/useReceiveAddress.tsx

```diff
@@ -11,29 +12,36 @@ import {
   selectBuyAsset,
   selectManualReceiveAddress,
 } from 'state/slices/swappersSlice/selectors'
-import { useAppDispatch, useAppSelector } from 'state/store'
+import { useAppSelector } from 'state/store'
 
-const getReceiveAddress = async ({
-  asset,
-  wallet,
-  accountMetadata,
-}: GetReceiveAddressArgs): Promise<string | undefined> => {
-  const { chainId } = fromAssetId(asset.assetId)
-  const { accountType, bip44Params } = accountMetadata
-  const chainAdapter = getChainAdapterManager().get(chainId)
-  if (!(chainAdapter && wallet)) return
-  const { accountNumber } = bip44Params
-  try {
-    return await chainAdapter.getAddress({ wallet, accountNumber, accountType })
-  } catch (e) {
-    console.log(e)
-  }
-}
+export const getReceiveAddress = pMemoize(
```

Comment:
This ran a *lot* more than necessary and was causing a bunch of spam (to the Ledger, but not only, this runs way too much regardless of the wallet) at best, but more often than not, actual bugs - think trying to run `useReceiveAddress` while having the wrong app opened

Reply:
Note:
1. This is now p-memoized, meaning rejections won't be cached so we can attempt again in case of a failed call and aren't rugged for the lifetime of the sesh
2. This is now exported for pure use outside of react hooks

Reply:
🧠